app/react/src/server/framework-preset-react.ts

```diff
@@ -20,13 +21,28 @@ export async function babel(config: TransformOptions, options: StorybookOptions)
     plugins: [require.resolve('react-refresh/babel'), ...(config.plugins || [])],
   };
 }
+const storybookReactDirName = path.dirname(require.resolve('@storybook/react/package.json'));
+// TODO: improve node_modules detection
+const context = storybookReactDirName.includes('node_modules')
+  ? path.join(storybookReactDirName, '../../') // Real life case, already in node_modules
+  : path.join(storybookReactDirName, '../../node_modules'); // SB Monorepo
+
+const hasJsxRuntime = () => {
+  try {
+    require.resolve('react/jsx-runtime', { paths: [context] });
```

Comment:
Why does it try to use node_modules folder for lookup? It can't work with yarn 2 pnp.

CRA uses `require.resolve('react/jsx-runtime');` for the detection:
https://github.com/facebook/create-react-app/blob/282c03f9525fdf8061ffa1ec50dce89296d916bd/packages/react-scripts/config/webpack.config.js#L79-L85